src/main/java/org/humancellatlas/ingest/notifications/sources/impl/rabbit/RabbitNotificationSource.java

```diff
@@ -0,0 +1,61 @@
+package org.humancellatlas.ingest.notifications.sources.impl.rabbit;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule;
+import java.io.IOException;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Stream;
+import lombok.RequiredArgsConstructor;
+import org.humancellatlas.ingest.messaging.Constants.Queues;
+import org.humancellatlas.ingest.notifications.model.Notification;
+import org.humancellatlas.ingest.notifications.sources.NotificationSource;
+import org.humancellatlas.ingest.notifications.sources.impl.inmemory.InmemoryNotificationSource;
+import org.springframework.amqp.rabbit.annotation.RabbitListener;
+import org.springframework.amqp.rabbit.core.RabbitMessagingTemplate;
+import org.springframework.stereotype.Component;
+
+@Component
+@RequiredArgsConstructor
+public class RabbitNotificationSource implements NotificationSource {
+  private final InmemoryNotificationSource inmemoryNotificationSource = new InmemoryNotificationSource();
+  private final RabbitMessagingTemplate rabbitMessagingTemplate;
+  private final AmqpConfig amqpConfig;
+
+  @RabbitListener(queues = Queues.NOTIFICATIONS_QUEUE)
+  private void listen(String notification) {
```

Comment:
It's weird to me that it has to listen to the rabbit queue when it's the one sending messages to it. Can we not fetch the messages directly from the Rabbit queues when we need to?

Reply:
The messages aren't being sent to the queue per se, they're sent to an exchange with a given routing key. This is with the intention that other interested consumers could see these notifications.

The stream() could be re-implemented as a Stream.generate() calling a poller on the rabbit queue. This would be a whole lot of extra code with no benefit though.